casual/casual-api/src/main/java/se/laz/casual/jca/CasualConnection.java

```diff
@@ -23,4 +23,20 @@ public interface CasualConnection extends CasualServiceApi, CasualQueueApi, Casu
      */
     @Override
     void close();
+
+    /**
+     * Get the current domain id for this connection
+     * @return the current domain id for this connection
+     */
+    DomainId getDomainID();
+
+    /**
+     * If a connection is gone, for whatever reason - it is just not usable anymore.
+     *
+     * When this happens, {@link ConnectionObserver#connectionGone(DomainId)} is called with the same domain id
+     * that can be retrieved via {@link CasualConnection#getDomainID()}
+     * @param observer - the observer that will be notified
+     */
+    void addConnectionObserver(ConnectionObserver observer);
```

Comment:
Do we need to be able to remove observers too?

Reply:
Not that I can see. A connection exists and can be used, by an observer, until notified that it can not be used anymore. That is the use case, it's gone thus both usage and observability.

Reply:
But who are the connection observers? Is this going to be casual caller?
Are the NettyNetworkConnection objects reused in the RA connection pool? If so, this could lead to the list of observers being populated with stale information as they are never cleared.
Symmetry gut feel, if you can request to observe something, you should also be able to request to no longer observe it too.

Reply:
Connection observer will be anyone who wants to make use of a connection and who wants to keep caching information re the connection. This can only be done per domain id, thus why it has to be exposed. 
Casual caller will want want to use this to do its caching.